server/api.js

```diff
@@ -60,6 +60,57 @@ const cantPlay = (user, tourney) =>
     "Mapper",
   ]);
 
+const canEditWarmup = async (user, playerNo, match) => {
+  async function isCaptainOf(playerName, teamName, tourney) {
+    const team = await Team.findOne({ name: teamName, tourney: tourney }).populate('players');
+    if (!team || !team.players || !team.players[0]) return false;
+    return team.players[0].username === playerName;
+  }
+
+  // Admin can always edit warmup
+  if (user.isAdmin) return true;
```

Comment:
this permission check is broken because `isAdmin` isn't a property of `user`
instead you can use `isAdmin(user, match.tourney)`

Reply:
Fixed in 622fc1fcce3125de39357266b81011d788216425 and b9b4a5df043a95e9a4108dd6dbefd11d6cf4f546